src/Downloading/AddAuthenticationHeader.php

```diff
@@ -27,9 +28,13 @@ public static function withAuthHeaderFromComposer(RequestInterface $request, Pac
         array_walk(
             $authHeaders,
             static function (string $v) use (&$request): void {
-                // @todo probably process this better
-                $headerParts = explode(':', $v);
-                $request     = $request->withHeader(trim($headerParts[0]), trim($headerParts[1]));
+                $headerParts = array_map('trim', explode(':', $v, 2));
```

Comment:
nit: `'trim'` can be replaced with `trim(...)`, first callable syntax is available since php 8.1

Reply:
That's true! But by using the "string syntax", the callable is resolved at the engine level instead of the language level which can be a bit faster. But if the project coding standards has a rule to use first class callable, let's update this part 🙂